core/include/detray/propagator/base_stepper.hpp

```diff
@@ -231,13 +193,6 @@ class base_stepper {
         free_matrix_type m_jac_transport =
             matrix_operator().template identity<e_free_size, e_free_size>();
 
-        /// Full jacobian
-        bound_matrix_type m_full_jacobian =
-            matrix_operator().template identity<e_bound_size, e_bound_size>();
```

Comment:
Are you considering that Jacobian needs to be fetched in kalman actor?

https://github.com/acts-project/traccc/blob/3350e5076c14398aa5a3daf17f11c0a1656936d3/core/include/traccc/fitting/kalman_filter/kalman_actor.hpp#L121

Reply:
Part of the idea of the PR was to make it impossible to fetch them from the stepper. Because otherwise the code could compile successfully without the ```parameter_transporter``` and the ```parameter_resetter``` being part of the actor chain. This might result in wrong results without throwing errors.

The same is done with the bound track parameters for the ```pointwise_material_interactor``` in this PR. Do you foresee a problem why this cannot be declared here https://github.com/acts-project/traccc/blob/3350e5076c14398aa5a3daf17f11c0a1656936d3/core/include/traccc/fitting/kalman_filter/kalman_fitter.hpp#L62 the same way?

Reply:
I did not read the code changes carefully - so we can use the transporter state in the kalman actor operator?

Meanwhile I suggest to rethink this idea. Currently, the parameter transport is designed for the action on the surface but we have not implemented the covariance transport in the volume material. I am not sure if we will be able to use parameter transporter for this purpose.

Reply:
> I did not read the code changes carefully - so we can use the transporter state in the kalman actor operator?

Exactly

> Meanwhile I suggest to rethink this idea. Currently, the parameter transport is designed for the action on the surface but we have not implemented the covariance transport in the volume material. I am not sure if we will be able to use parameter transporter for this purpose.

Oh, I thought this would mainly change how the ```q/p``` in the transport jacobian would be calculated in the in the stepper and the scattering/energy loss would be applied by a different material interactor? Would the stepper need the bound parameters or full jacobian to do any of this?

Reply:
To be clear, I am referring the multiple scattering in the volume material (We have transport jacobian update from the energy loss in the volume material, and  this does not require covariance and full jacobian)

> Would the stepper need the bound parameters or full jacobian to do any of this?

Covariance update from multiple scattering will need both covariance and full jacobian. What matters is where we actually do this update. 

- If we do the MS covariance update of volume material in the stepper, the stepper will need them
- If we do the MS covariance update of volume material in the parameter transporter, either stepper or transporter will need them.

Therefore, having them in the stepper is the safer option in my opinion. Transporter may be used for this purpose so I can not stop this PR strictly speaking. Just be aware that there will be updates on this...

Reply:
Hm, I was expecting, that we would get a ```continous_material_interactor``` or something like this? I understand that the material thickness parameter needs to be carefully integrated along the trajectory in the stepper, especially if we have inhomogeneous volume material. Not sure why the stepper will need to continuously update the covariance (mostly it will not be on surface and we don't have a free covariance). But I will be happy to change it back if the bound covariance and full jacobian will in the end need to be updated in the stepper

Reply:
> If we have inhomogeneous volume material. 

We might not have inhomogeneous volume material. I don't see a good reason to support this.

> Not sure why the stepper will need to continuously update the covariance

Because the MS is not represented by the Jacobian and its corresponding joint covariance ($C_{MS}$) should be added to the bound covariance ($C$). Check what pointwise material interactor does for this:

$C \leftarrow C + C_{MS}$

Reply:
> We might not have inhomogeneous volume material. I don't see a good reason to support this.

Then why do we need more than the exact pathlength through the volume from the stepper?

> Because the MS is not represented by the Jacobian and its corresponding joint covariance ( C M S ) should be added to the bound covariance ( C ). Check what pointwise material interactor does for this:
> 
> C ← C + C M S

Yes, I know, but in which way is this different for volume material?

The stepper does not know what the destination surface is: How can it update any bound covariance/parameters while stepping between surfaces?

If everything happens at the destination surface, like in the ```pointwise_material_interactor```, then I don't understand what the potential problem is, sorry?